application/common/dos.py

```diff
@@ -102,21 +103,30 @@ def is_valid(self) -> bool:
         return None not in (self.easting, self.northing, self.latitude, self.longitude)
 
 
-def get_matching_dos_services(odscode: str) -> List[DoSService]:
+def get_matching_dos_services(odscode: str, org_type_id: str) -> List[DoSService]:
     """Retrieves DoS Services from DoS database
 
     Args:
         odscode (str): ODScode to match on
+        org_type_id (str): OrganisationType to match on
 
     Returns:
         list[DoSService]: List of DoSService objects with matching first 5
         digits of odscode, taken from DoS database
     """
-
-    logger.info(f"Searching for DoS services with ODSCode that matches first 5 digits of '{odscode}'")
-
-    sql_query = f"SELECT {', '.join(DoSService.db_columns)} FROM services WHERE odscode LIKE %(ODS_5)s"
-    named_args = {"ODS_5": f"{odscode[0:5]}%"}
+    if org_type_id == PHARMACY_ORG_TYPE_ID:
```

Comment:
The only thing that really changes between this IF block and the ELIF block below it are the ODSCodes that are then put in the sql query. Would we be missing anything if only the ODSCode that is put in the sql query is created within the blocks and then a more general log message could be written and the sql query creation made outside to avoid duplicate lines.

Reply:
Correct me if I missed something that differs between them

Reply:
Yes, you are correct. the previous implementation was different. I will make these changes.